// permitio/build/main/openapi/api/condition-sets-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConditionSetsApi = exports.ConditionSetsApiFactory = exports.ConditionSetsApiFp = exports.ConditionSetsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ConditionSetsApi - axios parameter creator
 * @export
 */
exports.ConditionSetsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Creates a new condition set (can be either a user set or a resource set).
         * @summary Create Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetCreate} conditionSetCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createConditionSet: async (projId, envId, conditionSetCreate, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('createConditionSet', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('createConditionSet', 'envId', envId);
            // verify required parameter 'conditionSetCreate' is not null or undefined
            common_1.assertParamExists('createConditionSet', 'conditionSetCreate', conditionSetCreate);
            const localVarPath = `/v2/schema/{proj_id}/{env_id}/condition_sets`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(conditionSetCreate, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Deletes a condition set and all its related data. This includes any permissions granted to said condition set (i.e: any matching condition set rules).
         * @summary Delete Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteConditionSet: async (projId, envId, conditionSetId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('deleteConditionSet', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('deleteConditionSet', 'envId', envId);
            // verify required parameter 'conditionSetId' is not null or undefined
            common_1.assertParamExists('deleteConditionSet', 'conditionSetId', conditionSetId);
            const localVarPath = `/v2/schema/{proj_id}/{env_id}/condition_sets/{condition_set_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'condition_set_id'}}`, encodeURIComponent(String(conditionSetId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Gets a single condition set, if such condition set exists.
         * @summary Get Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getConditionSet: async (projId, envId, conditionSetId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('getConditionSet', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('getConditionSet', 'envId', envId);
            // verify required parameter 'conditionSetId' is not null or undefined
            common_1.assertParamExists('getConditionSet', 'conditionSetId', conditionSetId);
            const localVarPath = `/v2/schema/{proj_id}/{env_id}/condition_sets/{condition_set_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'condition_set_id'}}`, encodeURIComponent(String(conditionSetId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all condition sets matching a filter.
         * @summary List Condition Sets
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetType} [type] if provided, will return only the condition sets of the specified type. e.g: only user sets.
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listConditionSets: async (projId, envId, type, page, perPage, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('listConditionSets', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('listConditionSets', 'envId', envId);
            const localVarPath = `/v2/schema/{proj_id}/{env_id}/condition_sets`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (type !== undefined) {
                localVarQueryParameter['type'] = type;
            }
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (perPage !== undefined) {
                localVarQueryParameter['per_page'] = perPage;
            }
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Partially updates a condition set. Fields that will be provided will be completely overwritten.
         * @summary Update Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetUpdate} conditionSetUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateConditionSet: async (projId, envId, conditionSetId, conditionSetUpdate, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('updateConditionSet', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('updateConditionSet', 'envId', envId);
            // verify required parameter 'conditionSetId' is not null or undefined
            common_1.assertParamExists('updateConditionSet', 'conditionSetId', conditionSetId);
            // verify required parameter 'conditionSetUpdate' is not null or undefined
            common_1.assertParamExists('updateConditionSet', 'conditionSetUpdate', conditionSetUpdate);
            const localVarPath = `/v2/schema/{proj_id}/{env_id}/condition_sets/{condition_set_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'condition_set_id'}}`, encodeURIComponent(String(conditionSetId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PATCH' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(conditionSetUpdate, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
/**
 * ConditionSetsApi - functional programming interface
 * @export
 */
exports.ConditionSetsApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.ConditionSetsApiAxiosParamCreator(configuration);
    return {
        /**
         * Creates a new condition set (can be either a user set or a resource set).
         * @summary Create Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetCreate} conditionSetCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createConditionSet(projId, envId, conditionSetCreate, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createConditionSet(projId, envId, conditionSetCreate, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Deletes a condition set and all its related data. This includes any permissions granted to said condition set (i.e: any matching condition set rules).
         * @summary Delete Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteConditionSet(projId, envId, conditionSetId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteConditionSet(projId, envId, conditionSetId, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Gets a single condition set, if such condition set exists.
         * @summary Get Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getConditionSet(projId, envId, conditionSetId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getConditionSet(projId, envId, conditionSetId, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Lists all condition sets matching a filter.
         * @summary List Condition Sets
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetType} [type] if provided, will return only the condition sets of the specified type. e.g: only user sets.
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listConditionSets(projId, envId, type, page, perPage, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listConditionSets(projId, envId, type, page, perPage, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Partially updates a condition set. Fields that will be provided will be completely overwritten.
         * @summary Update Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetUpdate} conditionSetUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateConditionSet(projId, envId, conditionSetId, conditionSetUpdate, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateConditionSet(projId, envId, conditionSetId, conditionSetUpdate, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
/**
 * ConditionSetsApi - factory interface
 * @export
 */
exports.ConditionSetsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.ConditionSetsApiFp(configuration);
    return {
        /**
         * Creates a new condition set (can be either a user set or a resource set).
         * @summary Create Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetCreate} conditionSetCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createConditionSet(projId, envId, conditionSetCreate, options) {
            return localVarFp
                .createConditionSet(projId, envId, conditionSetCreate, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Deletes a condition set and all its related data. This includes any permissions granted to said condition set (i.e: any matching condition set rules).
         * @summary Delete Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteConditionSet(projId, envId, conditionSetId, options) {
            return localVarFp
                .deleteConditionSet(projId, envId, conditionSetId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Gets a single condition set, if such condition set exists.
         * @summary Get Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getConditionSet(projId, envId, conditionSetId, options) {
            return localVarFp
                .getConditionSet(projId, envId, conditionSetId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Lists all condition sets matching a filter.
         * @summary List Condition Sets
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetType} [type] if provided, will return only the condition sets of the specified type. e.g: only user sets.
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listConditionSets(projId, envId, type, page, perPage, options) {
            return localVarFp
                .listConditionSets(projId, envId, type, page, perPage, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Partially updates a condition set. Fields that will be provided will be completely overwritten.
         * @summary Update Condition Set
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} conditionSetId Either the unique id of the condition set, or the URL-friendly key of the condition set (i.e: the \&quot;slug\&quot;).
         * @param {ConditionSetUpdate} conditionSetUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateConditionSet(projId, envId, conditionSetId, conditionSetUpdate, options) {
            return localVarFp
                .updateConditionSet(projId, envId, conditionSetId, conditionSetUpdate, options)
                .then((request) => request(axios, basePath));
        },
    };
};
/**
 * ConditionSetsApi - object-oriented interface
 * @export
 * @class ConditionSetsApi
 * @extends {BaseAPI}
 */
class ConditionSetsApi extends base_1.BaseAPI {
    /**
     * Creates a new condition set (can be either a user set or a resource set).
     * @summary Create Condition Set
     * @param {ConditionSetsApiCreateConditionSetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ConditionSetsApi
     */
    createConditionSet(requestParameters, options) {
        return exports.ConditionSetsApiFp(this.configuration)
            .createConditionSet(requestParameters.projId, requestParameters.envId, requestParameters.conditionSetCreate, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes a condition set and all its related data. This includes any permissions granted to said condition set (i.e: any matching condition set rules).
     * @summary Delete Condition Set
     * @param {ConditionSetsApiDeleteConditionSetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ConditionSetsApi
     */
    deleteConditionSet(requestParameters, options) {
        return exports.ConditionSetsApiFp(this.configuration)
            .deleteConditionSet(requestParameters.projId, requestParameters.envId, requestParameters.conditionSetId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Gets a single condition set, if such condition set exists.
     * @summary Get Condition Set
     * @param {ConditionSetsApiGetConditionSetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ConditionSetsApi
     */
    getConditionSet(requestParameters, options) {
        return exports.ConditionSetsApiFp(this.configuration)
            .getConditionSet(requestParameters.projId, requestParameters.envId, requestParameters.conditionSetId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all condition sets matching a filter.
     * @summary List Condition Sets
     * @param {ConditionSetsApiListConditionSetsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ConditionSetsApi
     */
    listConditionSets(requestParameters, options) {
        return exports.ConditionSetsApiFp(this.configuration)
            .listConditionSets(requestParameters.projId, requestParameters.envId, requestParameters.type, requestParameters.page, requestParameters.perPage, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Partially updates a condition set. Fields that will be provided will be completely overwritten.
     * @summary Update Condition Set
     * @param {ConditionSetsApiUpdateConditionSetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ConditionSetsApi
     */
    updateConditionSet(requestParameters, options) {
        return exports.ConditionSetsApiFp(this.configuration)
            .updateConditionSet(requestParameters.projId, requestParameters.envId, requestParameters.conditionSetId, requestParameters.conditionSetUpdate, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.ConditionSetsApi = ConditionSetsApi;
