// permitio/build/main/openapi/api/opaldata-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OPALDataApi = exports.OPALDataApiFactory = exports.OPALDataApiFp = exports.OPALDataApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * OPALDataApi - axios parameter creator
 * @export
 */
exports.OPALDataApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Get All Data
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {boolean} [internalUpdateCache]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAllData: async (orgId, projId, envId, internalUpdateCache, options = {}) => {
            // verify required parameter 'orgId' is not null or undefined
            common_1.assertParamExists('getAllData', 'orgId', orgId);
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('getAllData', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('getAllData', 'envId', envId);
            const localVarPath = `/v2/internal/opal_data/{org_id}/{proj_id}/{env_id}`
                .replace(`{${'org_id'}}`, encodeURIComponent(String(orgId)))
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (internalUpdateCache !== undefined) {
                localVarQueryParameter['__internal_update_cache'] = internalUpdateCache;
            }
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get Data For Role
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} roleId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDataForRole: async (orgId, projId, envId, roleId, options = {}) => {
            // verify required parameter 'orgId' is not null or undefined
            common_1.assertParamExists('getDataForRole', 'orgId', orgId);
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('getDataForRole', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('getDataForRole', 'envId', envId);
            // verify required parameter 'roleId' is not null or undefined
            common_1.assertParamExists('getDataForRole', 'roleId', roleId);
            const localVarPath = `/v2/internal/opal_data/{org_id}/{proj_id}/{env_id}/roles/{role_id}`
                .replace(`{${'org_id'}}`, encodeURIComponent(String(orgId)))
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'role_id'}}`, encodeURIComponent(String(roleId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * return permission assignment data between user sets and resource sets (i.e: condition set rules data)
         * @summary Get Data For Set Rule
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} userSetId
         * @param {string} resourceSetId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDataForSetRule: async (orgId, projId, envId, userSetId, resourceSetId, options = {}) => {
            // verify required parameter 'orgId' is not null or undefined
            common_1.assertParamExists('getDataForSetRule', 'orgId', orgId);
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('getDataForSetRule', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('getDataForSetRule', 'envId', envId);
            // verify required parameter 'userSetId' is not null or undefined
            common_1.assertParamExists('getDataForSetRule', 'userSetId', userSetId);
            // verify required parameter 'resourceSetId' is not null or undefined
            common_1.assertParamExists('getDataForSetRule', 'resourceSetId', resourceSetId);
            const localVarPath = `/v2/internal/opal_data/{org_id}/{proj_id}/{env_id}/condition_set_rules/{user_set_id}/{resource_set_id}`
                .replace(`{${'org_id'}}`, encodeURIComponent(String(orgId)))
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'user_set_id'}}`, encodeURIComponent(String(userSetId)))
                .replace(`{${'resource_set_id'}}`, encodeURIComponent(String(resourceSetId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get Data For User
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} userId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDataForUser: async (orgId, projId, envId, userId, options = {}) => {
            // verify required parameter 'orgId' is not null or undefined
            common_1.assertParamExists('getDataForUser', 'orgId', orgId);
            // verify required parameter 'projId' is not null or undefined
            common_1.assertParamExists('getDataForUser', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            common_1.assertParamExists('getDataForUser', 'envId', envId);
            // verify required parameter 'userId' is not null or undefined
            common_1.assertParamExists('getDataForUser', 'userId', userId);
            const localVarPath = `/v2/internal/opal_data/{org_id}/{proj_id}/{env_id}/users/{user_id}`
                .replace(`{${'org_id'}}`, encodeURIComponent(String(orgId)))
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'user_id'}}`, encodeURIComponent(String(userId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
/**
 * OPALDataApi - functional programming interface
 * @export
 */
exports.OPALDataApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.OPALDataApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @summary Get All Data
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {boolean} [internalUpdateCache]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getAllData(orgId, projId, envId, internalUpdateCache, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getAllData(orgId, projId, envId, internalUpdateCache, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @summary Get Data For Role
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} roleId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getDataForRole(orgId, projId, envId, roleId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getDataForRole(orgId, projId, envId, roleId, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * return permission assignment data between user sets and resource sets (i.e: condition set rules data)
         * @summary Get Data For Set Rule
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} userSetId
         * @param {string} resourceSetId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getDataForSetRule(orgId, projId, envId, userSetId, resourceSetId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getDataForSetRule(orgId, projId, envId, userSetId, resourceSetId, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @summary Get Data For User
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} userId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getDataForUser(orgId, projId, envId, userId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getDataForUser(orgId, projId, envId, userId, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
/**
 * OPALDataApi - factory interface
 * @export
 */
exports.OPALDataApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.OPALDataApiFp(configuration);
    return {
        /**
         *
         * @summary Get All Data
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {boolean} [internalUpdateCache]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAllData(orgId, projId, envId, internalUpdateCache, options) {
            return localVarFp
                .getAllData(orgId, projId, envId, internalUpdateCache, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get Data For Role
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} roleId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDataForRole(orgId, projId, envId, roleId, options) {
            return localVarFp
                .getDataForRole(orgId, projId, envId, roleId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * return permission assignment data between user sets and resource sets (i.e: condition set rules data)
         * @summary Get Data For Set Rule
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} userSetId
         * @param {string} resourceSetId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDataForSetRule(orgId, projId, envId, userSetId, resourceSetId, options) {
            return localVarFp
                .getDataForSetRule(orgId, projId, envId, userSetId, resourceSetId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get Data For User
         * @param {string} orgId Either the unique id of the organization, or the URL-friendly key of the organization (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} userId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDataForUser(orgId, projId, envId, userId, options) {
            return localVarFp
                .getDataForUser(orgId, projId, envId, userId, options)
                .then((request) => request(axios, basePath));
        },
    };
};
/**
 * OPALDataApi - object-oriented interface
 * @export
 * @class OPALDataApi
 * @extends {BaseAPI}
 */
class OPALDataApi extends base_1.BaseAPI {
    /**
     *
     * @summary Get All Data
     * @param {OPALDataApiGetAllDataRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof OPALDataApi
     */
    getAllData(requestParameters, options) {
        return exports.OPALDataApiFp(this.configuration)
            .getAllData(requestParameters.orgId, requestParameters.projId, requestParameters.envId, requestParameters.internalUpdateCache, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get Data For Role
     * @param {OPALDataApiGetDataForRoleRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof OPALDataApi
     */
    getDataForRole(requestParameters, options) {
        return exports.OPALDataApiFp(this.configuration)
            .getDataForRole(requestParameters.orgId, requestParameters.projId, requestParameters.envId, requestParameters.roleId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * return permission assignment data between user sets and resource sets (i.e: condition set rules data)
     * @summary Get Data For Set Rule
     * @param {OPALDataApiGetDataForSetRuleRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof OPALDataApi
     */
    getDataForSetRule(requestParameters, options) {
        return exports.OPALDataApiFp(this.configuration)
            .getDataForSetRule(requestParameters.orgId, requestParameters.projId, requestParameters.envId, requestParameters.userSetId, requestParameters.resourceSetId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get Data For User
     * @param {OPALDataApiGetDataForUserRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof OPALDataApi
     */
    getDataForUser(requestParameters, options) {
        return exports.OPALDataApiFp(this.configuration)
            .getDataForUser(requestParameters.orgId, requestParameters.projId, requestParameters.envId, requestParameters.userId, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.OPALDataApi = OPALDataApi;
