// permitio/build/main/openapi/api/policy-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyApi = exports.PolicyApiFactory = exports.PolicyApiFp = exports.PolicyApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PolicyApi - axios parameter creator
 * @export
 */
exports.PolicyApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Get Opal Data Sources
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOpalDataSources: async (options = {}) => {
            const localVarPath = `/v2/opal/data/config`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get Opal Data Sources
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOpalDataSourcesOpalDataConfigGet: async (options = {}) => {
            const localVarPath = `/opal/data/config`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
/**
 * PolicyApi - functional programming interface
 * @export
 */
exports.PolicyApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.PolicyApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @summary Get Opal Data Sources
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOpalDataSources(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOpalDataSources(options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @summary Get Opal Data Sources
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOpalDataSourcesOpalDataConfigGet(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOpalDataSourcesOpalDataConfigGet(options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
/**
 * PolicyApi - factory interface
 * @export
 */
exports.PolicyApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.PolicyApiFp(configuration);
    return {
        /**
         *
         * @summary Get Opal Data Sources
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOpalDataSources(options) {
            return localVarFp.getOpalDataSources(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get Opal Data Sources
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOpalDataSourcesOpalDataConfigGet(options) {
            return localVarFp
                .getOpalDataSourcesOpalDataConfigGet(options)
                .then((request) => request(axios, basePath));
        },
    };
};
/**
 * PolicyApi - object-oriented interface
 * @export
 * @class PolicyApi
 * @extends {BaseAPI}
 */
class PolicyApi extends base_1.BaseAPI {
    /**
     *
     * @summary Get Opal Data Sources
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PolicyApi
     */
    getOpalDataSources(options) {
        return exports.PolicyApiFp(this.configuration)
            .getOpalDataSources(options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get Opal Data Sources
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PolicyApi
     */
    getOpalDataSourcesOpalDataConfigGet(options) {
        return exports.PolicyApiFp(this.configuration)
            .getOpalDataSourcesOpalDataConfigGet(options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.PolicyApi = PolicyApi;
