// permitio/build/module/config.js
import _ from 'lodash';
// returns a config
export class ConfigFactory {
    static defaults() {
        return {
            token: _.get(process.env, 'PERMIT_API_KEY', ''),
            pdp: _.get(process.env, 'PERMIT_PDP_URL', 'http://localhost:7766'),
            apiUrl: _.get(process.env, 'PERMIT_API_URL', 'https://api.permit.io'),
            log: {
                // log level, default is warn (warnings and errors)
                level: _.get(process.env, 'PERMIT_LOG_LEVEL', 'warn'),
                // Label added to logs
                label: _.get(process.env, 'PERMIT_LOG_LABEL', 'Permit.io'),
                // When logging - dump full data to console as JSON
                json: JSON.parse(_.get(process.env, 'PERMIT_LOG_JSON', 'false')),
            },
            multiTenancy: {
                defaultTenant: 'default',
                useDefaultTenantIfEmpty: true,
            },
            timeout: undefined,
            throwOnError: undefined,
        };
    }
    /**
     * Returns a valid config, for unset values returns the default config
     * @param options - a partial configuration
     * @returns
     */
    static build(options) {
        const config = _.merge(_.assign({}, ConfigFactory.defaults()), options);
        return config;
    }
}
