// permitio/build/module/enforcement/enforcer.js
import axios from 'axios';
import { ContextStore } from '../utils/context';
const RESOURCE_DELIMITER = ':';
function isString(x) {
    return typeof x === 'string';
}
export class PermitError extends Error {
    constructor(message) {
        super(message);
        this.name = 'PermitError';
    }
}
export class PermitConnectionError extends PermitError {
    constructor(message) {
        super(message);
        this.name = 'PermitConnectionError';
    }
}
export class PermitPDPStatusError extends PermitError {
    constructor(message) {
        super(message);
        this.name = 'PermitPDPStatusError';
    }
}
/**
 * this client is dealing with evaluation of check() queries.
 */
export class Enforcer {
    constructor(config, logger) {
        this.config = config;
        this.logger = logger;
        this.client = axios.create({
            baseURL: `${this.config.pdp}/`,
        });
        this.logger = logger;
        this.contextStore = new ContextStore();
    }
    /**
     * Usage:
     *
     * // with (resource, action):
     * const user = { key: 'UNIQUE_USER_ID' };
     * permit.check(user, 'get', {'type': 'task', 'key': '23'})
     * permit.check(user, 'get', {'type': 'task'})
     *
     * // with (url, method):
     * const { resource, action } = permit.getUrlContext('/lists/3/todos/37', 'GET');
     * permit.check(user, action, resource)
     *
     * @param user
     * @param action
     * @param resource
     * @param context
     * @param config
     *
     * @returns whether or not action is permitted for given user
     */
    async check(user, action, resource, context = {}, // context provided specifically for this query
    config = {}) {
        return await this.checkWithExceptions(user, action, resource, context, config).catch((err) => {
            const shouldThrow = config.throwOnError === undefined ? this.config.throwOnError : config.throwOnError;
            if (shouldThrow) {
                throw err;
            }
            else {
                this.logger.error(err);
                return false;
            }
        });
    }
    async checkWithExceptions(user, action, resource, context = {}, // context provided specifically for this query
    config = {}) {
        const normalizedUser = isString(user) ? { key: user } : user;
        const checkTimeout = config.timeout || this.config.timeout;
        const resourceObj = isString(resource) ? Enforcer.resourceFromString(resource) : resource;
        const normalizedResource = this.normalizeResource(resourceObj);
        const queryContext = this.contextStore.getDerivedContext(context);
        const input = {
            user: normalizedUser,
            action: action,
            resource: normalizedResource,
            context: queryContext,
        };
        return await this.client
            .post('allowed', input, {
            headers: {
                Authorization: `Bearer ${this.config.token}`,
            },
            timeout: checkTimeout,
        })
            .then((response) => {
            if (response.status !== 200) {
                throw new PermitPDPStatusError(`Permit.check() got an unexpected status code: ${response.status}, please check your SDK init and make sure the PDP sidecar is configured correctly. \n\
            Read more about setting up the PDP at https://docs.permit.io`);
            }
            const decision = ('allow' in response.data ? response.data.allow : response.data.result.allow) || false;
            this.logger.info(`permit.check(${Enforcer.userRepr(normalizedUser)}, ${action}, ${Enforcer.resourceRepr(resourceObj)}) = ${decision}`);
            return decision;
        })
            .catch((error) => {
            const errorMessage = `Error in permit.check(${Enforcer.userRepr(normalizedUser)}, ${action}, ${Enforcer.resourceRepr(resourceObj)})`;
            if (axios.isAxiosError(error)) {
                const errorStatusCode = error.response?.status.toString() || '';
                const errorDetails = error?.response?.data
                    ? JSON.stringify(error.response.data)
                    : error.message;
                this.logger.error(`[${errorStatusCode}] ${errorMessage}, err: ${errorDetails}`);
            }
            else {
                this.logger.error(`${errorMessage}\n${error}`);
            }
            throw new PermitConnectionError(`Permit SDK got error: \n ${error.message} \n
          and cannot connect to the PDP, please check your configuration and make sure the PDP is running at ${this.config.pdp} and accepting requests. \n
          Read more about setting up the PDP at https://docs.permit.io`);
        });
    }
    // TODO: remove this eventually, once we decide on finalized structure of AuthzQuery
    normalizeResource(resource) {
        const normalizedResource = Object.assign({}, resource);
        // if tenant is empty, we might auto-set the default tenant according to config
        if (!normalizedResource.tenant && this.config.multiTenancy.useDefaultTenantIfEmpty) {
            normalizedResource.tenant = this.config.multiTenancy.defaultTenant;
        }
        return normalizedResource;
    }
    static userRepr(user) {
        if (user.attributes || user.email) {
            return JSON.stringify(user);
        }
        return user.key;
    }
    static resourceRepr(resource) {
        if (resource.attributes && resource.attributes.length > 0) {
            return JSON.stringify(resource);
        }
        let resourceRepr = '';
        if (resource.tenant) {
            resourceRepr += `${resource.tenant}/`;
        }
        resourceRepr += `${resource.type}:${resource.key ?? '*'}`;
        return resourceRepr;
    }
    static resourceFromString(resource) {
        const parts = resource.split(RESOURCE_DELIMITER);
        if (parts.length < 1 || parts.length > 2) {
            throw Error(`permit.check() got invalid resource string: '${resource}'`);
        }
        return {
            type: parts[0],
            key: parts.length > 1 ? parts[1] : undefined,
        };
    }
    getMethods() {
        return {
            check: this.check.bind(this),
        };
    }
}
