// permitio/build/main/config.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigFactory = void 0;
const lodash_1 = __importDefault(require("lodash"));
// returns a config
class ConfigFactory {
    static defaults() {
        return {
            token: lodash_1.default.get(process.env, 'PERMIT_API_KEY', ''),
            pdp: lodash_1.default.get(process.env, 'PERMIT_PDP_URL', 'http://localhost:7766'),
            apiUrl: lodash_1.default.get(process.env, 'PERMIT_API_URL', 'https://api.permit.io'),
            log: {
                // log level, default is warn (warnings and errors)
                level: lodash_1.default.get(process.env, 'PERMIT_LOG_LEVEL', 'warn'),
                // Label added to logs
                label: lodash_1.default.get(process.env, 'PERMIT_LOG_LABEL', 'Permit.io'),
                // When logging - dump full data to console as JSON
                json: JSON.parse(lodash_1.default.get(process.env, 'PERMIT_LOG_JSON', 'false')),
            },
            multiTenancy: {
                defaultTenant: 'default',
                useDefaultTenantIfEmpty: true,
            },
            timeout: undefined,
            throwOnError: undefined,
        };
    }
    /**
     * Returns a valid config, for unset values returns the default config
     * @param options - a partial configuration
     * @returns
     */
    static build(options) {
        const config = lodash_1.default.merge(lodash_1.default.assign({}, ConfigFactory.defaults()), options);
        return config;
    }
}
exports.ConfigFactory = ConfigFactory;
