// permitio/build/main/logger.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoggerFactory = exports.prettyConsoleLog = void 0;
const util_1 = __importDefault(require("util"));
const winston_1 = __importDefault(require("winston"));
function prettyConsoleLog(label, data) {
    console.log(label, util_1.default.inspect(data, false, 12, true));
}
exports.prettyConsoleLog = prettyConsoleLog;
const FgGreen = '\x1b[32m';
const FgCyan = '\x1b[36m';
// const COLOR_RESET: string = "\x1b[0m";
// const FgBlack: string = "\x1b[30m";
// const FgRed: string = "\x1b[31m";
// const FgYellow: string = "\x1b[33m";
// const FgBlue: string = "\x1b[34m";
// const FgMagenta: string = "\x1b[35m";
// const FgWhite: string = "\x1b[37m";
const consoleFormat = winston_1.default.format.printf(({ level, message, label }) => {
    return `${FgCyan}${label} [${level}]: ${FgGreen} ${message}`;
});
const MESSAGE = Symbol.for('message');
const LEVEL = Symbol.for('level');
class SimpleConsoleTransport extends winston_1.default.transports.Console {
    log(info, callback) {
        setImmediate(() => this.emit('logged', info));
        if (this.stderrLevels[info[LEVEL]]) {
            console.error(info[MESSAGE]);
        }
        else {
            console.log(info[MESSAGE]);
        }
        if (callback) {
            callback();
        }
    }
}
class LoggerFactory {
    static createLogger(config) {
        return winston_1.default.createLogger({
            level: config.log.level,
            format: winston_1.default.format.simple(),
            transports: [
                new SimpleConsoleTransport({
                    format: winston_1.default.format.combine(winston_1.default.format.label({ label: config.log.label }), winston_1.default.format.timestamp(), config.log.json ? winston_1.default.format.prettyPrint() : consoleFormat, winston_1.default.format.colorize({ all: true })),
                }),
            ],
        });
    }
}
exports.LoggerFactory = LoggerFactory;
